src/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -742,6 +744,27 @@ def file_exists(self, filepath: _PATH, trainer: "pl.Trainer") -> bool:
         exists = self._fs.exists(filepath)
         return trainer.strategy.broadcast(exists)
 
+    def _should_remove_checkpoint(self, trainer: "pl.Trainer", previous: str, current: str) -> bool:
+        """Checks if the previous checkpoint should be deleted.
+
+        A checkpoint won't be deleted if it is a local file and any of the following cases apply:
+        - The previous checkpoint is the same as the current checkpoint
```

Comment:
This is also a condition when it is not a local file. The docstring makes it sound like this only applies to local files

Reply:
Thanks for spotting it. I refined the description